client/packages/system/src/Encounter/DetailView/DetailView.tsx

```diff
@@ -76,14 +76,17 @@ export const DetailView: FC = () => {
     isError,
   } = useEncounter.document.byId(id);
 
+  const isVaccinationEncounter =
+    encounter?.programEnrolment?.isImmunisationProgram;
+
   // If this is a vaccination encounter, we want to use the suggested
   // next vaccination dates for the next encounter
   const {
     query: { data: vaccineCard },
   } = usePatientVaccineCard(encounter?.programEnrolment?.id ?? '');
-  const suggestedNextEncounterDate = getNextVaccinationEncounterDate(
-    vaccineCard?.items ?? []
-  );
+  const suggestedNextEncounterDate = isVaccinationEncounter
+    ? getNextVaccinationEncounterDate(vaccineCard?.items ?? [])
+    : null;
```

Comment:
Not a necessary change, in that for non-immunisation encounters, vaccineCard would come back empty anyway... but it did feel like it was worth being explicit, currently we only return suggested encounter date for vaccination encounters...